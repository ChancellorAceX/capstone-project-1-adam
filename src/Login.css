* {
  box-sizing: border-box;
}

.container {
  height:96vh;
  width:97vw;
  display:flex;
  justify-content: center;
  align-items: center;
}

form {
  width:fit-content;
  padding:15px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  border:1px black solid;
}

label,input {
  display:block;
  margin-bottom: 10px;
}

button {
  width:
}