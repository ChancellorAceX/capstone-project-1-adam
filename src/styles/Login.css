main.loginContainer {
  position:fixed;
  top:10vh;
  background-color: black;
  color:#aaaaaa;
  height:82vh;
  width:100vw;
  display:flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
} 
 
form.loginForm {
  padding:5vw;
  text-align: center;
  max-width:300px;
}

form.loginForm * {
  margin-bottom: 5vh;
}
 
p.login {
  text-align: center;
}

form.loginForm div {
  color:#ba0404;
  font-weight: 800;
}

button.loginSubmit {
  font-weight: 600;
  width:45vw;
  max-width:200px;
  border-radius:20px;
  background-color: #b40a0a;
  color: #aaaaaa;
  text-shadow:-1px -1px 0 #444444, 1px 1px 0 #444444,1px -1px 0 #444444, -1px 1px 0 #444444;
}