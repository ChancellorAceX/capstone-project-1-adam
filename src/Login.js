import React from 'react';
import './Login.css';

export default function Login(){
  return(
    <section className='container'>
      <form>
        <h2>Application Name</h2>
        <label for='username'>Username:</label>
        <input type='text' id='username' placeholder='awesomesauce'/>
        <label for='password'>Password:</label>
        <input type='text' id='password' placeholder='inorite?'></input>
        <button type='submit'>Submit</button>
      </form>
    </section>
  )
}