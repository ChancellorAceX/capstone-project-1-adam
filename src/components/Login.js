import {React} from 'react';
import { withRouter } from 'react-router';
import { Link } from 'react-router-dom';
import AuthApiService from '../services/auth-api-service';
import TokenService from '../services/token-service';
import '../styles/Login.css';
import Header from './Header';
import config from '../config';

export default withRouter(class Login extends React.Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => { },
    },
  };
  state = { error: null };

  handleLoginSuccess = () => {
    fetch(`${config.API_ENDPOINT}/api/encounter`, {
      method: 'GET',
      headers: {
        'content-type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      }
    })
      .then(res => res.json())
      .then(encounters => this.props.setEncounters(encounters))
      .catch(error => console.log({ error }));
    this.props.loggedInToggle(true);
    const { location, history } = this.props;
    const destination = (location.state || {}).from || '/encounters';
    history.push(destination);


  };

  handleSubmitJwtAuth = ev => {
    ev.preventDefault();
    this.setState({ error: null });
    const { username, password } = ev.target;
    AuthApiService.postLogin({
      username: username.value,
      password: password.value
    })
      .then(res => {
        username.value = '';
        password.value = '';
        TokenService.saveAuthToken(res.authToken);
        this.handleLoginSuccess();
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };
  render() {
    const { error } = this.state;
    return (
      <main className='loginContainer'>
        <Header
          pageName='Log In'
          loggedInToggle={this.props.loggedInToggle}
          history={this.props.history}
        />
        <form className='loginForm' onSubmit={this.handleSubmitJwtAuth}>
          <div role='alert'>
            {error && <p className='red'>{error}</p>}
          </div>
          <h2>Log in to see your campaigns!</h2>
          <label htmlFor='username'>Username:</label>
          <input type='text' id='username' placeholder='username' />
          <label htmlFor='password'>Password:</label>
          <input type='password' id='password' placeholder='password'></input>
          <button className='loginSubmit' type='submit'>Submit</button>
        </form>
        <p className='login'>New User? <Link to='/register'>Register Here!</Link></p>
        <p>Just curious? Enter username: usera, password: usera, to have a look!</p>
      </main>
    );
  }
});