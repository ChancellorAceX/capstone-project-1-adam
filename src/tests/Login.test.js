import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from '../components/Login';

describe('Login component', () => {
const props = {
  "setEncounters": "ƒ () {}",
  "history": {
    "length": 10,
    "action": "PUSH",
    "location": "{hash: \"\", key: \"y3xvl7\", pathname: \"/login\", searc…}",
    "createHref": "ƒ createHref() {}",
    "push": "ƒ push() {}",
    "replace": "ƒ replace() {}",
    "go": "ƒ go() {}",
    "goBack": "ƒ goBack() {}",
    "goForward": "ƒ goForward() {}",
    "block": "ƒ block() {}",
    "listen": "ƒ listen() {}"
  },
  "loggedInToggle": "ƒ () {}",
  "location": {
    "pathname": "/login",
    "search": "",
    "hash": "",
    "key": "y3xvl7"
  },
  "match": {
    "path": "/login",
    "url": "/login",
    "isExact": true,
    "params": "{}"
  }
};

  it('renders the full login component', () => {
    const wrapper = shallow(<Login {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});