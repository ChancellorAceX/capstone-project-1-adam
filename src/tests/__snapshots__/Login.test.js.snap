// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders the full login component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
